.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,24 @@
+{
+	"name": "pyhf-devcontainer",
+	"build": {
+		"context": "..",
+		"dockerfile": "../docker/Dockerfile"
+	},
+	"features": {
+		"ghcr.io/devcontainers/features/git:1": {}
+	},
+
+	"customizations": {
+		"vscode": {
+			"settings": {
+				"python.defaultInterpreterPath": "/opt/venv/bin/python"
+			},
+			"extensions": [
+				"ms-python.python",
+				"ms-toolsai.jupyter",
+				"GitHub.codespaces",
+				"ms-vscode-remote.remote-containers"
```

Comment:
I'll let @crazy4pi314 give the final answer here, but my understanding is that those extensions are not actually necessary here, unless you want to connect to a remote container / codespace from _inside_ the dev container (of course the extensions should not hurt either).

Reply:
> unless you want to connect to a remote container / codespace

While the pyhf dev team doesn't actively use Codespaces at the moment (very cool, but not clear if the cost is worth it atm for us) we would like for others to use Codespaces or GitPod for demos and to help people making contributions get onboarded. Do these extensions help / become needed there?

Reply:
Yeah I think they are probably not needed, Its likely my fault that I was exporting envs and those got installed as needed (but you don't have to proactively)